services/management/enclave/src/service.rs

```diff
@@ -255,9 +255,10 @@ impl TeaclaveManagement for TeaclaveManagementService {
     ) -> TeaclaveServiceResponseResult<RegisterFunctionResponse> {
         let user_id = self.get_request_user_id(request.metadata())?;
 
-        let function = Function::from(request.message)
-            .id(Uuid::new_v4())
-            .owner(user_id);
+        let mut function = Function::from(request.message);
+        // TODO: use setter functions
+        function.id = Uuid::new_v4();
+        function.owner = user_id;
```

Comment:
Could it use `FunctionBuilder` here?

Reply:
The `From` trait  has a commnet `We explicitly construct Function here in case of missing any field` with it, so I left the `from` here. It is not trivial to add setter/getter functions to all the strutures. Besides, it is not very neat to have many setter/getter functions. I do not have a very good idea now. Maybe more `From` traits are needed? Therefore, I added a `TODO` there.

Reply:
Thank you for the explaination! Maybe a `From<RegisterFunctionRequest>` for `FunctionBuilder` can solve this problem and making the call to builders more consistent.